File: lab4/src/data/FileReader.java
package data;

import java.io.File;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class FileReader {

    public static List<Flat> readFlats(String fileName) {
        List<Flat> flats = new ArrayList<>();

        try (Scanner scanner = new Scanner(new File(fileName))) {
            while (scanner.hasNextLine()) {
                int roomsCount = scanner.nextInt();
                int floor = scanner.nextInt();
                double area = scanner.nextDouble();
                String district = scanner.next();

                Flat flat = new Flat(district, roomsCount, floor, area);
                flats.add(flat);
            }
        } catch (Exception e) {
            System.err.println(e.getMessage());
        }

        return flats;
    }
}

File: lab6/src/data/FileReader.kt
package data

import java.io.File
import java.util.Scanner

object FileReader {

    fun readFlats(fileName: String): List<Flat> {
        val flats = mutableListOf<Flat>()

        try {
            val file = File(fileName)
            file.forEachLine { line: String ->
                val scanner = Scanner(line)
                val roomsCount = scanner.nextInt()
                val floor = scanner.nextInt()
                val area = scanner.nextDouble()
                val district = scanner.next()

                val flat = Flat(district, roomsCount, floor, area)
                flats.add(flat)
            }
        } catch (e: Exception) {
            System.err.println(e)
        }

        return flats
    }
}
